Sources/XiCliCore/CommandLineTool.swift

```diff
@@ -0,0 +1,155 @@
+// Copyright 2018 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Cocoa
+
+public final class CommandLineTool {
+    private let arguments: [String]
+    
+    public init(arguments: [String] = CommandLine.arguments) {
+        self.arguments = Array(arguments.dropFirst())
+    }
+    
+    public func run() throws {
+        if arguments.isEmpty {
+            help()
+            return
+        }
+        
+        let args = try getArguments(from: arguments)
+        
+        if let args = args {
+            let filePath = try resolvePath(from: args.fileInput)
+            try openFile(at: filePath)
+            
+            if args.wait {
+                print("waiting for editor to close...")
+                let group = DispatchGroup()
+                group.enter()
+                setObserver(group: group, filePath: filePath)
+                group.wait()
+                print("closed")
```

Comment:
we can probably lose this `print` message? I think it's helpful to show when we're waiting, but I think it will be obvious when we're finished.

Reply:
Removed it!